packages/xstate-vue/src/useMachine.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  ref,
+  Ref,
+  watch,
+  onBeforeMount,
+  onBeforeUnmount
+} from '@vue/composition-api';
+import {
+  interpret,
+  EventObject,
+  StateMachine,
+  State,
+  Interpreter,
+  InterpreterOptions,
+  MachineOptions,
+  StateConfig
+} from 'xstate';
+
+interface UseMachineOptions<TContext, TEvent extends EventObject> {
+  /**
+   * If provided, will be merged with machine's `context`.
+   */
+  context?: Partial<TContext>;
+  /**
+   * The state to rehydrate the machine to. The machine will
+   * start at this state instead of its `initialState`.
+   */
+  state?: StateConfig<TContext, TEvent>;
+}
+
+const defaultOptions = {
```

Comment:
I don't think it gets actually used anyhow in the code.

Reply:
Yeah, you're right. 

Should I add the same implementation like [in the react hook?](https://github.com/davidkpiano/xstate/blob/master/packages/xstate-react/src/index.ts#L110)

Reply:
I don't think so - actually we need to remove `immediate` from the react hook as it's not async-react safe.